spring-boot-modules/spring-boot-libraries-2/src/main/java/com/baeldung/openapi/controller/ChatController.java

```diff
@@ -0,0 +1,44 @@
+package com.baeldung.openapi.controller;
+
+import com.baeldung.openapi.dto.ChatRequest;
+import com.baeldung.openapi.dto.ChatResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+@RestController
+public class ChatController {
+
+    @Qualifier("openaiRestTemplate")
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${openai.model}")
+    private String model;
+
+    @Value("${openai.api.url}")
+    private String apiUrl;
+
+    @GetMapping("/chat")
+    public String chat(@RequestParam String prompt) {
+        // create a request
+        ChatRequest request = new ChatRequest(model, prompt);
+
+        // call the API
+        ChatResponse response = restTemplate.postForObject(
+                apiUrl,
+                request,
+                ChatResponse.class);
+
+        if (response == null || response.getChoices() == null || response.getChoices().isEmpty()) {
+            return "No response";
+        }
+
+        // return the first response
+        return response.getChoices().get(0).getMessage().getContent();
```

Comment:
Will be great to remove single comments and replace them with javadoc

Reply:
Updated